### PR TITLE
feat(webdriver): add support for selenium webdriver proxy

### PR DESCRIPTION
**Pull request by vliloia**
Allows the configuration of a proxy by using selenium webdriver `Builder#usingWebDriverProxy()` (was added with [v2.46.0](https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/CHANGES.md#v2460)).

This is useful when you sit behind a corporate proxy and you need to connect to a remote webdriver server (e.g. BrowserStack).

There is also an issue related to it #2345.

### CODE REVIEW

**Comment by vliloia** [comment]
`I signed it!`

File: lib/driverProviders/driverProvider.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ DriverProvider.prototype.getExistingDrivers = function() {
 DriverProvider.prototype.getNewDriver = function() {
   var builder = new webdriver.Builder().
     usingServer(this.config_.seleniumAddress).
+    usingWebDriverProxy(this.config_.seleniumProxy).
```
Comment (juliemr) [inline lib/driverProviders/driverProvider.js]: Can you add a note to https://github.com/angular/protractor/blob/master/docs/referenceConf.js explaining what this config option does?

Comment (juliemr) [inline lib/driverProviders/driverProvider.js]: Could we name it `webDriverProxy`? Unless there's a reason to have a discrepancy between the function name and the config name.

**Comment by vliloia** [comment]
Thanks for reviewing. Added changes

**Comment by juliemr** [comment]
Thanks! Merged as aa5ceb576f0283b6591faaa95e342ef3c603c717

**Pull request closed without merging**